src/runtime/arc.ts

```diff
@@ -138,14 +141,24 @@ export class Arc implements ArcInterface {
     return this._loader;
   }
 
+  set modality(modality: Modality) {
+    this._modality = modality;
+  }
+
   get modality(): Modality {
-    if (this.peh.slotComposer && this.peh.slotComposer.modality) {
-      return this.peh.slotComposer.modality;
+    let modalities = [];
+    if (this._modality) {
+      modalities.push(this._modality);
     }
+    // TODO(sjmiles): Modality rules are unclear. Seems to me the Arc should declare it's own modality
```

Comment:
this might be a bit ad hoc, but here is how i was thinking about it originally.
`modality` is the maximal possible set of modalities that can be supported for this arc:
1. in an actually running arc, where there is an actual slot-composer, it dictates the modality (because only recipes supporting these modalities can be instantiated)
2. in an arc with an active recipe (that was instantiated, but possible not currently running) - the modalities are ones that can support this recipe
3. in an empty arc, any modality supporting a recipe in a context is good
(so it kinda narrows down when going from 3 to 1).

Reply:
Thanks for the notes.

For posterity: all Arcs have a (monomorphic) SlotComposer instance now. I believe we can keep the same concept, just use the term `SlotObserver` now, which is now the thing that talks to actual devices.